### PR TITLE
BUG: Cannot map MultiIndex to more levels

### LINKED ISSUE #24800: df.index.map with difference size fails for Pandas > 0.22
#### Code Sample

```python
import pandas as pd

df = pd.DataFrame({'a': [0,1,2,3],
                   'b': ['a_1_bar', 'a_2_bar', 'b_1_bar', 'b_2_bar'],
                   'c': list('defg')})

df = df.set_index(['b','c'])
df.index.map(lambda x: tuple(x[0].split('_')))
```
#### Problem description

The code above works in Pandas 0.22 and lower, but fails since 0.23. This seems to be due to the fact that Pandas wants to preserve the names of the levels in the old index.

When the amount of levels in the new index is different compared to the old one, this fails with  a `ValueError` because of this mismatch.

`ValueError: Length of names must match number of levels in MultiIndex.`

#### Expected Output

Older version of pandas returned a new MultiIndex, without names for the levels.

```python
MultiIndex(levels=[['a', 'b'], ['1', '2'], ['bar']],
           labels=[[0, 0, 1, 1], [0, 1, 0, 1], [0, 0, 0, 0]])
```

I'm not sure whether this change was deliberate. If not, a workaround might be to only preserve the names if the new amount of levels matches the old one. And otherwise disregard the names, resulting in similar behavior as in the olders Pandas version.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.6.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.23.4
pytest: 4.0.2
pip: 18.1
setuptools: 40.5.0
Cython: None
numpy: 1.15.4
scipy: 1.1.0
pyarrow: None
xarray: 0.11.2
IPython: 7.1.1
sphinx: 1.8.2
patsy: None
dateutil: 2.7.5
pytz: 2018.7
blosc: None
bottleneck: None
tables: None
numexpr: 2.6.8
feather: None
matplotlib: 3.0.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.6.3
html5lib: 1.0.1
sqlalchemy: 1.2.14
pymysql: None
psycopg2: 2.7.6.1 (dt dec pq3 ext lo64)
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### LINKED ISSUE #47173: BUG: Cannot map MultiIndex to more levels
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [x] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

df = pd.DataFrame({
    'a': [0, 1, 2, 3],
}, index=['a_1/1', 'a_2/2', 'b_1/1', 'b_2/2'])

# split index into multiindex
df.index = df.index.map(lambda x: tuple(x.split('_')))
print(df.index)

# MultiIndex([('a', '1/1'),
#             ('a', '2/2'),
#             ('b', '1/1'),
#             ('b', '2/2')],
#            )

# cannot split the created multiindex into more levels
# ValueError: Length of names must match number of levels in MultiIndex.
df.index.map(lambda x: (x[0], *x[1].split('/')))
```

### Issue Description

An `Index` can be mapped into a `MultiIndex` with any number of levels. However, mapping an `MultiIndex` into more levels does not work. Applying `map` on a `MultiIndex` requires that the return value is either a single element (`Index`) or a tuple with the same number of elements as before. Returning a tuple with more elements than before is not possible.

### Expected Behavior

Expected would be a `MultiIndex` with more levels than the previous one:

```python
import pandas as pd
df = pd.DataFrame({
    'a': [0, 1, 2, 3],
}, index=['a_1/1', 'a_2/2', 'b_1/1', 'b_2/2'])

# split index into multiindex
df.index = df.index.map(lambda x: tuple(x.split('_')))
print(df.index)

#MultiIndex([('a', '1/1'),
#            ('a', '2/2'),
#            ('b', '1/1'),
#            ('b', '2/2')],
#           )

# circumvents the problem by flattening the index first
df.index = df.index.to_flat_index().map(lambda x: (x[0], *x[1].split('/')))
print(df.index)

#MultiIndex([('a', '1', '1'),
#           ('a', '2', '2'),
#            ('b', '1', '1'),
#            ('b', '2', '2')],
#           )



```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : c7f7443c1bad8262358114d5e88cd9c8a308e8aa
python           : 3.8.8.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : AMD64 Family 23 Model 113 Stepping 0, AuthenticAMD
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : de_DE.cp1252

pandas           : 1.3.1
numpy            : 1.21.4
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 21.0.1
setuptools       : 59.2.0
Cython           : 0.29.23
pytest           : 6.2.5
hypothesis       : None
sphinx           : 4.1.2
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.3
html5lib         : 1.1
pymysql          : 1.0.2
psycopg2         : 2.9.1 (dt dec pq3 ext lo64)
jinja2           : 3.0.3
IPython          : 7.20.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : None
fsspec           : 2021.10.0
fastparquet      : None
gcsfs            : None
matplotlib       : 3.5.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 6.0.0
pyxlsb           : None
s3fs             : None
scipy            : 1.6.3
sqlalchemy       : 1.4.13
tables           : None
tabulate         : 0.8.9
xarray           : None
xlrd             : None
xlwt             : None
numba            : 0.53.0

</details>

### PR DESCRIPTION
**Pull request by GPla**
- [x] closes #24800 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -6364,7 +6364,7 @@ def map(self, mapper, na_action=None):
 
         # we can return a MultiIndex
         if new_values.size and isinstance(new_values[0], tuple):
-            if isinstance(self, MultiIndex):
+            if isinstance(self, MultiIndex) and len(new_values[0]) == self.nlevels:
```
Comment (jreback) [inline pandas/core/indexes/base.py:6367]: what happens if its a Multiindex bu tthis is not true? do we have a test case for that

Comment (GPla) [reply pandas/core/indexes/base.py:6367]: The checks that happen here are only to determine the `names` of the new index. 

```python
if isinstance(self, MultiIndex) and len(new_values[0]) == self.nlevels:
     names = self.names
 elif self.name:
    names = [self.name] * len(new_values[0])
else:
  names = None
```

If the old index is a `MultiIndex` then we reuse  `names` for the new index. However, if the number of levels changed then `len(self.names) != len(len(new_values[0])` and an error will be raised. With the change, `names` is not preserved and reset as seen in the snippet. The last of the three tests that I added checks for the new case.

File: doc/source/whatsnew/v1.5.0.rst [hunk 2]
```diff
@@ -802,6 +802,7 @@ MultiIndex
 - Bug in :meth:`Series.to_numpy` where multiindexed Series could not be converted to numpy arrays when an ``na_value`` was supplied (:issue:`45774`)
 - Bug in :class:`MultiIndex.equals` not commutative when only one side has extension array dtype (:issue:`46026`)
 - Bug in :meth:`MultiIndex.from_tuples` cannot construct Index of empty tuples (:issue:`45608`)
+- Bug in :meth:`Index.map` cannot map an existing :class:`MultiIndex` to more levels (:issue:`47173`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.5.0.rst:805]:
```suggestion
- Bug in :meth:`Index.map` cannot map an existing :class:`MultiIndex` to more levels (:issue:`24800`)
```

**Comment by GPla** [comment]
@simonjayhawkins I updated the code to reflect [#24800 (comment)](https://github.com/pandas-dev/pandas/issues/24800#issuecomment-455069121).

Furthermore, I removed the name preservation completely. Keeping the other check.

```python
if self.name:
    names = [self.name] * len(new_values[0])
else:
    names = None
```

would set `names` the first time we switch from an `Index` to a `MultiIndex` but would be lost in successive `map` operations since `MultiIndex` does not set/have a `name` property. Thus, I removed this check as well.

**Comment by simonjayhawkins** [comment]
> EDIT: There are 5 failing tests which expect a name preservation from `MultiIndex` to `MultiIndex` and one test which expects a name preservation from `Index` to `MultiIndex`. How should this be resolved?

we may need to deprecate the existing behavior. cc @jorisvandenbossche @toobaz  (from #24800 discussion)

**Pull request closed without merging**